ledger/ledger-api-integration-tests/src/test/itsuite/scala/com/digitalasset/platform/tests/integration/ledger/api/DivulgenceIT.scala

```diff
@@ -139,26 +142,38 @@ class DivulgenceIT
           )
         )
       )
+
+      transaction <- transactionClient(ctx)
+        .getTransactions(
+          ledgerEndBeforeSubmission,
+          None,
+          TransactionFilter(Map(party -> Filters.defaultInstance)))
+        .filter(_.commandId == commandId)
+        .runWith(Sink.head)
     } yield ()
 
   // here div1Cid is _divulged_ to Bob after it is created. the checks below
   // check that this divulgence is not visible in the ACS / flat transaction stream,
   // but that it is visible in the transaction trees.
   case class Setup(div1Cid: String, div2Cid: String)
 
-  private def createDivulgence1(ctx: LedgerContext): Future[ContractIdString] =
+  private val runSuffix: UUID = UUID.randomUUID
```

Comment:
At some point, we should factor out all ID mangling into some helper class or scalatest trait. See also https://github.com/digital-asset/daml/pull/1735.

Reply:
Yes definitely

Reply:
I am doing this as part of #1735